sensor/tests/resource/helper.go

```diff
@@ -295,6 +295,32 @@ func (c *TestContext) LastResourceStateWithTimeout(matchResourceFn MatchResource
 	}
 }
 
+// WaitForDeploymentEvent waits until sensor process a given deployment
+func (c *TestContext) WaitForDeploymentEvent(name string) {
+	c.WaitForDeploymentEventWithTimeout(name, 3*time.Second)
+}
+
+// WaitForDeploymentEventWithTimeout waits until sensor process a given deployment
+func (c *TestContext) WaitForDeploymentEventWithTimeout(name string, timeout time.Duration) {
+	timer := time.NewTimer(timeout)
+	ticker := time.NewTicker(500 * time.Millisecond)
```

Comment:
10ms was making the tests burn too much CPU ultimately making them run slower to the point that in some cases they were reaching the 10min timeout. I think the 10ms ticker was left out from #3302 by mistake.

I've created the constant `defaultTicker` so that can be reused in these functions as you suggested.